### ISSUE TITLE
Reconstruction error is increasing on every iteration for PyTorch PARAFAC decomposition

### ISSUE THREAD

**Issue by offendo** [tensorly/tensorly#148]
Hello,

#### Describe the bug

When running PARAFAC with a PyTorch backend on a tensor, the reconstruction error increases on every iteration, but only with certain tensors and only with missing values. If you don't include `mask` in the snippet below, the code works as expected. Also, if you don't scale the values to be between 1 and 5, the code works as expected (in my limited testing).

#### Steps or Code to Reproduce

```python
import tensorly.decomposition as tld
import tensorly as tl
import torch
import numpy as np

def tl_pytorch(tens, mask, rank):
    tl.set_backend('pytorch')
    tens = torch.DoubleTensor(tens)
    mask = torch.DoubleTensor(mask)
    factors = tld.parafac(
        tens,
        mask=mask,
        tol=10**-6,
        rank=rank,
        init='random',
        verbose=1,
        n_iter_max=30,
        return_errors=True,
    )
    return factors

tens = np.random.rand(50, 50, 50)  # Generates random 50x50x50 tensor between 0 and 1
mask = (tens > .9).astype(int)  # Uses 90% as missing data
tens = tens * mask  # makes all missing values = 0
tens = 4 * tens + 1  # scales all values to be between 1 and 5
rank = 10  # rank
factors = tl_pytorch(tens, mask, rank)  # calls parafac
```

#### Expected behavior

Reconstruction error should be decreasing.
 
#### Actual result
Reconstruction error increases: 
```bash
reconstruction error=0.7019533161420678
iteration 1,  reconstraction error: 1.2483085339787325, decrease = -0.5463552178366647, unnormalized = 791.2979992256775
iteration 2,  reconstraction error: 1.5861791789772561, decrease = -0.3378706449985236, unnormalized = 1005.4729071967671
iteration 3,  reconstraction error: 1.8188189570050688, decrease = -0.23263977802781266, unnormalized = 1152.9423715822843
iteration 4,  reconstraction error: 1.9840963249991213, decrease = -0.16527736799405246, unnormalized = 1257.7110622153068
iteration 5,  reconstraction error: 2.103536238405534, decrease = -0.1194399134064128, unnormalized = 1333.423565922177
iteration 6,  reconstraction error: 2.1909567653307174, decrease = -0.08742052692518332, unnormalized = 1388.8391031585272
iteration 7,  reconstraction error: 2.255654748740151, decrease = -0.06469798340943367, unnormalized = 1429.8509070774237
iteration 8,  reconstraction error: 2.3040345706405922, decrease = -0.04837982190044121, unnormalized = 1460.518690907031
iteration 9,  reconstraction error: 2.3405748586731434, decrease = -0.03654028803255116, unnormalized = 1483.6814395579029
iteration 10,  reconstraction error: 2.368443589534941, decrease = -0.02786873086179753, unnormalized = 1501.34732132642
iteration 11,  reconstraction error: 2.389903660132901, decrease = -0.02146007059796018, unnormalized = 1514.9507778959935
iteration 12,  reconstraction error: 2.406585946557338, decrease = -0.01668228642443692, unnormalized = 1525.5256153747473
iteration 13,  reconstraction error: 2.419675771837099, decrease = -0.01308982527976088, unnormalized = 1533.8232054914106
iteration 14,  reconstraction error: 2.430041642369828, decrease = -0.010365870532729193, unnormalized = 1540.3940911254592
iteration 15,  reconstraction error: 2.438324955412023, decrease = -0.008283313042194873, unnormalized = 1545.644851541522
iteration 16,  reconstraction error: 2.4450030360545747, decrease = -0.006678080642551709, unnormalized = 1549.8780612867731
iteration 17,  reconstraction error: 2.4504337826597027, decrease = -0.00543074660512799, unnormalized = 1553.3205907624329
iteration 18,  reconstraction error: 2.4548875259033114, decrease = -0.004453743243608699, unnormalized = 1556.1438015487115
iteration 19,  reconstraction error: 2.4585699301157526, decrease = -0.0036824042124412593, unnormalized = 1558.4780634770166
iteration 20,  reconstraction error: 2.4616385730048327, decrease = -0.003068642889080042, unnormalized = 1560.4232644528745
iteration 21,  reconstraction error: 2.464215032245444, decrease = -0.0025764592406112286, unnormalized = 1562.0564720988111
iteration 22,  reconstraction error: 2.4663937563213745, decrease = -0.0021787240759305604, unnormalized = 1563.4375569470046
iteration 23,  reconstraction error: 2.4682486182224364, decrease = -0.0018548619010618772, unnormalized = 1564.6133468027963
iteration 24,  reconstraction error: 2.469837788437615, decrease = -0.0015891702151784592, unnormalized = 1565.6207157163867
iteration 25,  reconstraction error: 2.47120738100391, decrease = -0.001369592566295097, unnormalized = 1566.488895199234
iteration 26,  reconstraction error: 2.472394198229574, decrease = -0.0011868172256641607, unnormalized = 1567.2412140936033
iteration 27,  reconstraction error: 2.4734278092401305, decrease = -0.0010336110105564345, unnormalized = 1567.8964161549272
iteration 28,  reconstraction error: 2.474332133204976, decrease = -0.0009043239648454104, unnormalized = 1568.4696636530862
iteration 29,  reconstraction error: 2.475126652114267, decrease = -0.0007945189092910532, unnormalized = 1568.9733061470338
```

#### Versions

``` python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import tensorly; print("TensorLy", tensorly.__version__)
```
```
Linux-5.3.0-arch1-1-ARCH-x86_64-with-arch
Python 3.7.4 (default, Oct  4 2019, 06:57:26)
[GCC 9.2.0]
NumPy 1.17.4
SciPy 1.3.3
TensorLy 0.4.5
```

**Comment by JeanKossaifi** [comment]
When solving the optimization problem, you may have an ill-posed problem to begin with (you chose a very low-rank compared to the size of the tensor), not to mention that 90% missing data, so it is possible that the algo does not converge. 

 In general it is good practice to normalise the input data, as you pointed out. I will add this to the documentation.
